uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Mail/MessageTest.php

```diff
@@ -0,0 +1,208 @@
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_HRLeaveAndAbsences_Factory_RequestNotificationTemplate as RequestNotificationTemplateFactory;
+
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Mail_MessageTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_Mail_MessageTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_LeaveRequestHelpersTrait;
+  use CRM_HRLeaveAndAbsences_LeaveManagerHelpersTrait;
+  use CRM_HRLeaveAndAbsences_MessageHelpersTrait;
+
+
+  private $leaveRequestTemplateFactory;
+
+  private $leaveContact;
+
+  public function setUp() {
+    CRM_Core_DAO::executeQuery('SET foreign_key_checks = 0;');
+    $this->leaveRequestTemplateFactory = new RequestNotificationTemplateFactory();
+
+    $this->leaveContact = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Staff1', 'last_name' => 'Staff1'], 'staffmember@dummysite.com'
+    );
+
+    $this->leaveRequestStatuses = $this->getLeaveRequestStatuses();
+    $this->leaveRequestDayTypes = $this->getLeaveRequestDayTypes();
+  }
+
+  public function testGetRecipientEmailsReturnsCorrectlyWhenLeaveContactHasLeaveApprover() {
+    $manager1 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Manager1', 'last_name' => 'Manager1'], 'manager1@dummysite.com'
+    );
+    $manager2 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Manager2', 'last_name' => 'Manager2'], 'manager2@dummysite.com'
+    );
+    $manager3 = ContactFabricator::fabricateWithEmail([
+      'first_name' => 'Manager3', 'last_name' => 'Manager3'], 'manager3@dummysite.com'
+    );
+
+    $this->setLeaveApproverRelationshipTypes(['has Leaves Approved By']);
+
+    // Set manager1 and manager2 only to be leave aprovers for the leave contact
+    $this->setContactAsLeaveApproverOf($manager1, $this->leaveContact, null, null, true, 'has Leaves Approved By');
+    $this->setContactAsLeaveApproverOf($manager2, $this->leaveContact, null, null, true, 'has Leaves Approved By');
+
+    $leaveRequest = LeaveRequest::create([
+      'type_id' => 1,
+      'contact_id' => $this->leaveContact['id'],
+      'status_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('tomorrow'),
+      'from_date_type' => 1,
+      'to_date' => CRM_Utils_Date::processDate('tomorrow'),
+      'to_date_type' => 1,
+      'toil_to_accrue' => 2,
+      'toil_duration' => 120,
+      'request_type' => LeaveRequest::REQUEST_TYPE_TOIL
+    ], false);
+
+    $message = new Message($leaveRequest, $this->leaveRequestTemplateFactory);
+    $recipientEmails = $message->getRecipientEmails($leaveRequest);
+
+    $this->assertCount(3, $recipientEmails);
+
+    //The contact and the leave approvers are eligible recipients for this email notification.
+    foreach($recipientEmails as $value) {
+      $this->assertContains($value['email'], [
+        'staffmember@dummysite.com', 'manager1@dummysite.com', 'manager2@dummysite.com'
+      ]);
```

Comment:
There's a problem with this assertion. If all the e-mail on the recipients is the same (which is a bug), the test will still pass. 

A safer way to handle this would be:
- get all the emails (probably using array_column())
- sort them, so we can always have them in an expected order
- finally compare the array of expected emails with the array of returned emails

This applies to all the other tests where you're doing similar checks